x-pack/plugins/security_solution/public/explore/components/risk_score/translations.ts

```diff
@@ -41,22 +49,24 @@ export const ENTITY_RISK_LEVEL = (riskEntity: RiskScoreEntity) =>
   });
 
 export const getRiskEntityTranslation = (
-  riskEntity: RiskScoreEntity,
+  riskEntity?: RiskScoreEntity,
   lowercase = false,
   plural = false
 ) => {
-  if (lowercase) {
-    if (plural) {
-      return (riskEntity === RiskScoreEntity.host ? HOSTS : USERS).toLowerCase();
-    }
+  const text = (() => {
+    if (!riskEntity && plural) return ENTITIES;
+    if (!riskEntity) return ENTITY;
+
+    if (riskEntity === RiskScoreEntity.host && plural) return HOSTS;
+    if (riskEntity === RiskScoreEntity.host) return HOST;
+
+    if (riskEntity === RiskScoreEntity.user && plural) return USERS;
+    if (riskEntity === RiskScoreEntity.user) return USER;
 
-    return (riskEntity === RiskScoreEntity.host ? HOST : USER).toLowerCase();
-  }
-  if (plural) {
-    return riskEntity === RiskScoreEntity.host ? HOSTS : USERS;
-  }
+    return '';
+  })();
```

Comment:
Can we extract this in a separate function instead of creating this IIFE on every call? 
Also, we don't use `if` without opening `{ ... }` keys, there's no explicit lint rule, but you won't see any in Kibana, we should keep code consistent.

What about something like this:

```
const getRiskEntityTranslationText = (
  riskEntity: RiskScoreEntity | undefined,
  plural: boolean
): string => {
  if (!riskEntity) {
    return plural ? ENTITIES : ENTITY;
  }
  if (riskEntity === RiskScoreEntity.host) {
    return plural ? HOSTS : HOST;
  }
  if (riskEntity === RiskScoreEntity.user) {
    return plural ? USERS : USER;
  }
  return '';
};
```

And then just:
```
export const getRiskEntityTranslation = (
  riskEntity?: RiskScoreEntity,
  lowercase = false,
  plural = false
): string => {
  const text = getRiskEntityTranslationText(riskEntity, plural);
  return lowercase ? text.toLowerCase() : text;
};
```

Reply:
Moved it from a IIFE to a standalone named function, thanks! not really sure what the standard practice regarding these is 😅 
IMO it's a bit more readable without the braces but it makes no actual difference. If we can introduce lint rules for it then yay


I'd rather not do nested checks though, especially when mixing different control flow constructs. 
Keeping it linear with short circuit returns makes it more flexible and can easily account for any random edge cases that might pop up. 
I'm also not too fond of implicitly handling the `null` but I guess that's ok. I put up a reworked version, let me know what you think.

Reply:
Yeah, thanks for addressing the comments.

I don't have anything against `if -> return` without braces, I was using it before I joined Elastic, but I was told the same thing I am telling you in my first PR 😂 . I agree with you that it should be controlled by the linter, but probably adding [the curly rule](https://eslint.org/docs/latest/rules/curly) implies a big refactor. 
BTW, Here's [the styleguide](https://github.com/elastic/kibana/blob/main/STYLEGUIDE.mdx), there's no explicit rule for this, but you'll see all `if -> return` statements have braces.

Besides all that, I still prefer the `switch` over the `if -> return`s approach, mainly because we check the `riskEntity` and `plural` only once (the `null` is handled implicitly on both approaches), but I won't be blocking the PR asking to use `switch` over `ifs`, that's a personal preference. 

However, I think we should use braces to be consistent with the rest of the application, so using `ifs` we could do:

```
if (riskEntity === RiskScoreEntity.host) {
  return plural ? HOSTS : HOST;
}
[...]
```

Reply:
Man, that was dumb, I meant to add the braces, sorry about that

But you're saying, if I keep the `if`s, I should still do the ternaries in each block?
The nesting is what I'd like to avoid, but if that's standard then I'll change it

Reply:
I used ternaries to avoid nesting, but if you want to nest an `if (plural) { ... } else { ... }` inside each entity condition, that's okay.

Reply:
Ah I see! 
Right, by nesting I meant we still do a control flow check inside each if block or case statement, not that we have nested code blocks.

But it seems it's pretty standard at Elastic to use ternaries in this fashion so I will go with that 👍🏽 
And in that case, the switch does seem a bit better to me, so I added your suggestion!

Thanks for taking the time to explain everything, really appreciate the help 🎉